gossip/server.go

```diff
@@ -125,7 +125,10 @@ func (s *server) jitteredGossipInterval() time.Duration {
 // then begins processing connecting clients in an infinite select
 // loop via goroutine. Periodically, clients connected and awaiting
 // the next round of gossip are awoken via the conditional variable.
-func (s *server) start(rpcServer *rpc.Server) {
+func (s *server) start(rpcServer *rpc.Server, stopper *util.Stopper) {
+	stopper.AddWorker()
+	defer stopper.SetStopped()
```

Comment:
Calling AddWorker and SetStopped in the same function is generally racy; we should move the call to AddWorker() to just before the `go s.start()` statement (it's important to be consistent here even if it's ok in this case, to make sure that the calls always match up).